src/tui/mod.rs

```diff
@@ -90,7 +92,7 @@ pub async fn run_tui(
                     .ok_or_else(|| miette!("No more crossterm events"))?
                     .into_diagnostic()
                     .wrap_err("Failed to get next crossterm event")?;
-                handle_event(&mut tui, event)?;
+                handle_event(&mut tui, event, terminal.get_frame())?;
```

Comment:
You should pass the `Rect` representing the size of the `Frame`, rather than the entire `Frame` (which has explicit lifetimes, access to the terminal `Buffer`, etc.).

```suggestion
                handle_event(&mut tui, event, terminal.get_frame().size())?;
```